### ISSUE TITLE
Change to EncryptedSharedPreferences for Android?

### ISSUE THREAD

**Issue by TehNrd** [martinkasa/capacitor-secure-storage-plugin#13]
Disclaimer, I'm fairly new to Android and Capacitor development.

For Android, is there a reason [EncryptedSharedPreferences](https://developer.android.com/topic/security/data) aren't used? It seems like this would greatly simplify the plugin and essentially remove the need for the entire `PasswordStorageHelper.java` class.

**Comment by martinkasa** [comment]
There is no any specific reason. As is stated in README, code for android is based of https://www.apriorit.com/dev-blog/432-using-androidkeystore
. If we want to change the way passwords are stored, then we would need to create some migration process, so users do not lost their data.

**Comment by TehNrd** [comment]
True, the migration piece probably doesn't make this worth the effort. I appreciate the response. I might take a stab and creating a similar plugin as I need to learn this stuff anyway.

**Comment by TehNrd** [comment]
Hey @martinkasa , I'd like to reopen this discussion and take a stab at delivering this feature. I see two main advantages.

The first being it would greatly simplify the encryption process for Android and use the standard encryption methods/practices provided by Android.

The second and primary driver is that it would also allow for much easier access to the storage if you need to access values from a background process. For example, 90% of the app might be an Ionic/Capacitor app, but perhaps there are some background processes that need to get an acessToken/refreshToken for to sync background data with an external API.

Implementation seems pretty straightforward but curious your thoughts on the best way to run a migration? Or perhaps not run a migration and have a flag to determine what time of encryption process is used.

**Comment by SavageCore** [comment]
I would say on data access perform the migration, if you `set` an existing key it would remove from the old store then add to Encrypted, similarily if you `get` the data it would be moved. That way the API stays the same and the transition is seamless.
